### ISSUE TITLE
Are object IDs required to be unique?

### ISSUE THREAD

**Issue by hombit** [lincc-frameworks/hyrax#460]
In many training datasets, it is hard to define a unique object ID for each training sample, especially if data is being generated or augmented. In [Uncle Val](https://github.com/lincc-frameworks/uncle-val), my single model input is a "source" (LSST terminology, aka photometric measurement), which frequently doesn't have a pre-defined unique ID, while using a 64-bit hash/random function would definitely cause collisions on 10-100 billion observations.

**Comment by drewoldag** [comment]
We discussed this in person, but for your use case, the data that is input to the model is required to have a unique identifier, but the 2d array of data that is provided to the loss function is _not_ required to have unique identifiers (or any identifiers at all). 

I think we can close this out with the answer, "yes", and the understanding that Hyrax isn't requiring additional unique ids in your particular case.
